docs/content/config/security/ssl.md

```diff
@@ -158,6 +158,143 @@ Certbot provisions certificates to `/etc/letsencrypt`. Add a volume to store the
+
+    ```yaml
+    services:
+      mailserver:
+        environments:
+          # Set SSL certificate type.
+          - SSL_TYPE=letsencrypt
+        volumes:
+          # Mount the cert folder generated by Certbot into mail-server:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro
+
+      certbot-cloudflare:
+        image: certbot/dns-cloudflare:latest
+        command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.example.com
+        volumes:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/
+          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
+        secrets:
+          - cloudflare-api-token
+
+    # Docs: https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose
+    # WARNING: In compose configs without swarm, the long syntax options have no effect,
+    # Ensure that you properly `chmod 600` and `chown 0:0` the file on disk. Effectively treated as a bind mount.
+    secrets:
+      cloudflare-api-token:
+        file: ./docker-data/certbot/secrets/cloudflare.ini
+    ```
+
+    Alternative using the `docker run` command (`secrets` feature is not available):
+
+      ```sh
+      docker run \
+        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/secrets/:/tmp/secrets/certbot/"
+        certbot/dns-cloudflare \
+        certonly --dns-cloudflare --dns-cloudflare-credentials /tmp/secrets/certbot/cloudflare.ini -d mail.example.com
+      ```
+
+6. Run the service to provision a certificate:
+
+    ```sh
+    docker-compose run certbot-cloudflare
+    ```
+
+7. You should see the following log output:
+
+    ```log
+    Saving debug log to /var/log/letsencrypt/letsencrypt. log | Requesting a certificate for mail.example.com
+    Waiting 10 seconds for DNS changes to propagate
+    Successfully received certificate.
+    Certificate is saved at: /etc/letsencrypt/live/mail.example.com/fullchain.pem
+    Key is saved at: /etc/letsencrypt/live/mail.example.com/privkey.pem
+    This certificate expires on YYYY-MM-DD.
+    These files will be updated when the certificate renews.
+    NEXT STEPS:
+    - The certificate will need to be renewed before it expires. Certbot can automatically renew the certificate in background, but you may need to take steps to enable that functionality. See https://certbot.org/renewal structions.
+    ```
+
+!!! tip "(Optional) Renewing certificate"
+
+    Since the above example only implements how to provision a certificate, since generated certificate will expire in 90 days, you might want to implement renew feature here.
+
+    In the following example, we will modify our created `docker-compose.yml` and create a container which will use for renewing the certificate
+
+    First, we modify our `docker-compose.yml` to add a service called `certbot-cloudflare-renew`:
```

Comment:
```suggestion
    We've only demonstrated how to provision a certificate, but it will expire in 90 days and need to be renewed before then.

    In the following example, add a new service (`certbot-cloudflare-renew`) into `docker-compose.yml` that will handle certificate renewals:
```